crates/cheatnet/src/execution/contract_execution_syscall_handler.rs

```diff
@@ -56,14 +41,10 @@ impl HintProcessorLogic for ContractExecutionSyscallHandler<'_> {
             ..
         })) = maybe_extended_hint
         {
-            let selector = &selector.value.to_bytes_be().1;
-            let selector = std::str::from_utf8(selector).unwrap();
-            let inputs = match extract_input(vm, input_start, input_end) {
-                Ok(inputs) => inputs,
-                Err(err) => return Err(err),
-            };
+            let selector = parse_selector(selector)?;
+            let inputs = extract_input(vm, input_start, input_end)?;
 
-            match selector {
+            match selector.as_str() {
```

Comment:
Wouldn't be better to return a string from `parse_selector`? I think in both cases we use `selector` as a string